templates/base.html

```diff
@@ -8,7 +8,12 @@
     <body>
         <div>
             <a href="/"> Homepage </a>
-            <a href="contact/"> Contact </a>
+            <a href="/contact"> Contact </a>
+            {% if user.is_authenticated %}
+                <a href="/users/logout"> Logout </a>
+            {% else%}
+                <a href="/users/login"> Login </a>
```

Comment:
Please define route namespaces and names instead of using directly the route path.